realm/realm-library/src/androidTestObjectServer/java/io/realm/SyncUserTests.java

```diff
@@ -74,6 +84,36 @@ public void currentUser_returnsNullIfUserExpired() {
         assertNull(SyncUser.currentUser());
     }
 
+    @Test
+    public void currentUser_throwsIfMultipleUsersLoggedIn() {
+        AuthenticationServer originalAuthServer = SyncManager.getAuthServer();
+        AuthenticationServer authServer = Mockito.mock(AuthenticationServer.class);
+        SyncManager.setAuthServerImpl(authServer);
```

Comment:
I think this line should be moved in the `try` block below to avoid a chance for mock `AuthenticationServer` to be left.